google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -476,6 +477,18 @@ static String getServiceAccountProjectId(String credentialsPath) {
     return project;
   }
 
+  /* 
+   * Returns true if the projectId is valid.
+   * projectId must be between 6 and 30 characters
+   * projectId can have lowercase letters, digits or hyphens
+   * and must start with a lowercase letter
+   */
+  private boolean isValidProjectId(String projectId) {
+    Pattern p = Pattern.compile("^[a-z][a-z0-9-]*$");
```

Comment:
I think a `static final` member might be more appropriate here.

Reply:
Just to make sure, the signature should be `static final boolean ...`, not `private static final boolean ...`, is that right?